### PR TITLE
docs: node api

### PR DESCRIPTION
**Pull request by SyMind**
## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/rspack/src/index.ts [hunk 1]
```diff
@@ -10,4 +10,4 @@ const rspack: Rspack = fn;
 
 export * from "./exports";
 export { rspack };
-module.exports = rspack;
```
Comment (h-a-n-a) [inline packages/rspack/src/index.ts:13]: This is not intentional, right?

File: website/docs/zh/api/node-api.mdx [hunk 2]
```diff
@@ -1,13 +1,317 @@
...
+    }
+
+    // Log result...
+  }
+);
+```
+
+## 自定义文件系统
+
+默认情况下，Rspack 使用普通文件系统来读取文件并将文件写入磁盘。但是，还可以使用不同类型的文件系统（内存, webDAV 等）来更改输入或输出行为。为了实现这一点，可以改变 `inputFileSystem` 或 `outputFileSystem`。例如，可以使用 [`memfs`](https://github.com/streamich/memfs) 替换默认的 `outputFileSystem`，以将文件写入到内存中，而不是写入到磁盘：
```
Comment (hardfist) [inline website/docs/zh/api/node-api.mdx:294]: 自定义inputFileSystem好像还没生效？ cc @h-a-n-a

Comment (hardfist) [reply website/docs/zh/api/node-api.mdx:294]: see https://github.com/web-infra-dev/rspack/issues/5091

Comment (SyMind) [reply website/docs/zh/api/node-api.mdx:296]: I've added a highlight to the `Custom File Systems` section.

File: website/docs/zh/api/node-api.mdx [hunk 3]
```diff
@@ -1,13 +1,319 @@
...
+  // 之后读取输出：
+  const content = fs.readFileSync('...');
+  compiler.close((closeErr) => {
+    // ...
+  });
+});
+```
+
+:::tip 提示
+你指定的输出文件系统需要 兼容 Node 自身的 `fs` 模块接口，接口需要提供 `mkdirp` 和 `join` 工具方法。
```
Comment (h-a-n-a) [inline website/docs/zh/api/node-api.mdx:318]: `mkdirp` 应该是不需要的：https://github.com/web-infra-dev/rspack/blob/fb4d5aa21c49a822b5184ddd30f3f14b4bc18dcc/packages/rspack/src/fileSystem.ts#L23

Comment (SyMind) [reply website/docs/zh/api/node-api.mdx]: 我看 join 也是不需要的，所以我移除了这个提示，并将这个不同增加到了『与 webpack 的差异』中。

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**